packages/react/src/errorboundary.tsx

```diff
@@ -0,0 +1,124 @@
+import * as Sentry from '@sentry/browser';
+import * as hoistNonReactStatic from 'hoist-non-react-statics';
+import * as React from 'react';
+
+export const UNKNOWN_COMPONENT = 'unknown';
+
+export type ErrorBoundaryProps = {
+  showDialog?: boolean;
+  dialogOptions?: Sentry.ReportDialogOptions;
+  fallback?: React.ReactNode;
+  renderKey?: string | number;
+  fallbackRender?(fallback: {
+    error: Error | null;
+    componentStack: string | null;
+    resetError(): void;
+  }): React.ReactNode;
+  onError?(error: Error, componentStack: string): void;
+  onMount?(): void;
+  onReset?(error: Error | null, componentStack: string | null): void;
+  onUnmount?(error: Error | null, componentStack: string | null): void;
+};
+
+type ErrorBoundaryState = {
+  componentStack: string | null;
+  error: Error | null;
+};
+
+const INITIAL_STATE = {
+  componentStack: null,
+  error: null,
+};
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  public state: ErrorBoundaryState = INITIAL_STATE;
+
+  public componentDidCatch(error: Error, { componentStack }: React.ErrorInfo): void {
+    Sentry.captureException(error, { contexts: { react: { componentStack } } });
+    const { onError, showDialog, dialogOptions } = this.props;
+    if (onError) {
+      onError(error, componentStack);
+    }
+    if (showDialog) {
+      Sentry.showReportDialog(dialogOptions);
+    }
+    this.setState({ error, componentStack });
+  }
+
+  public componentDidMount(): void {
+    const { onMount } = this.props;
+    if (onMount) {
+      onMount();
+    }
+  }
+
+  // If render key changes and there is an error, the component is reset.
+  // This provides an easy way for users to reset their error boundary.
+  public componentDidUpdate(prevProps: ErrorBoundaryProps): void {
+    const { error } = this.state;
+    const { renderKey, onReset } = this.props;
+    if (error !== null && !Object.is(renderKey, prevProps.renderKey)) {
```

Comment:
Is renderKey necessary? Could we get the same behavior by simply using `key`?

Reply:
Good point, people can just use `key` for this.